### PR TITLE
PYTHON-4807 - Specify how to handle unacknowledged+(ordered|verbose|m…

### PR DESCRIPTION
**Pull request by NoahStapp**
…ulti-batch) bulk writes

### CODE REVIEW

**Comment by blink1073** [comment]
Needs some `None` checks:

```
"Optional[WriteConcern]" has no attribute "acknowledged"  [union-attr]
            if not write_concern.acknowledged and verbose_results:
```

**Comment by NoahStapp** [comment]
> Needs some `None` checks:
> 
> ```
> "Optional[WriteConcern]" has no attribute "acknowledged"  [union-attr]
>             if not write_concern.acknowledged and verbose_results:
> ```

Whoops :yikes:

**Comment by blink1073** [comment]
Needs the prose test: https://github.com/mongodb/specifications/commit/c84488cb9cf185e31869547a6378a701124b4665#diff-9de75f337c20e76dc80a4e133a38583f9249c04f3348591c61cda4b393c74655R699

**Review by blink1073 — APPROVED** [review]
LGTM!

File: pymongo/asynchronous/mongo_client.py [hunk 1]
```diff
@@ -2354,6 +2354,13 @@ async def bulk_write(
             if not write_concern:
                 write_concern = self.write_concern
 
+        if write_concern and not write_concern.acknowledged and verbose_results:
+            raise InvalidOperation(
+                "Cannot request unacknowledged write concern and verbose results"
+            )
+        elif write_concern and not write_concern.acknowledged and ordered:
+            raise InvalidOperation("Cannot request unacknowledged write concern and ordered writes")
```
Comment (ShaneHarvey) [inline pymongo/asynchronous/mongo_client.py:2362]: Shouldn't there be some dead code we can remove now that we don't support these operations?

**Pull request merged**